### ISSUE TITLE
hub create ORGANIZATION/NAME cannot set team_id

### ISSUE THREAD

**Issue by hkdobrev** [github/hub#431]
``` bash
$ hub create ORGANIZATION/NAME
Error creating repository: Unprocessable Entity (HTTP 422)
Invalid value for "team_id": ""
```

I know the team_id which should be assigned to the new repo, but I cannot set it using `hub create`.

**Comment by scamden** [comment]
+1, would be amazing to specify team_id and better yet set a default team @mislav it looked from the readme that you guys have the go impl going?

**Comment by mislav** [comment]
> it looked from the readme that you guys have the go impl going?

@scamden Yes, hub is now entirely in Go (except for some test setup which still runs on Ruby)

**Comment by scamden** [comment]
@mislav you mentioned you would revisit this after go release, any thoughts yet?

**Comment by mislav** [comment]
@scamden Yes we need to revisit this. It's in our 2.3 milestone, and we'll get to it eventually.

**Comment by scamden** [comment]
awesome good to hear

On 29 June 2015 at 15:11, Mislav Marohnić <notifications@github.com> wrote:

> @scamden <https://github.com/scamden> Yes we need to revisit this. It's
> in our 2.3 milestone, and we'll get to it eventually.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/github/hub/issues/431#issuecomment-116861539>.
>



-- 
*Sterling Camden*
Software Engineer
502 Emerson Street | Palo Alto, CA 94301
303.520.2968 | sterling@relateiq.com

**Comment by joaosa** [comment]
@mislav Sounds like a good solution. Any news? :)

**Comment by mislav** [comment]
@joaosa No code yet to solve this issue, sorry. I accept contributions. Although, I'm not sure in a scenario where you belong to multiple teams, which team should be picked as a default team to create the repo in?

**Comment by scamden** [comment]
@mislav at least being able to set a team would be huge regardless of the default, but maybe for default team you could ask and store it in a config file or something?

**Comment by mislav** [comment]
@scamden Agreed. I think being able to set a team explicitly is more important feature than configuring a default team.

**Comment by hkdobrev** [comment]
@mislav What do you think of my earlier comment? 

> @mislav Sorry to bother, but any news on this one? Should something be changed in hub or this should be taken care of in the GitHub API? Thanks!

I think this still stands. You don't need to choose a team when creating an organisation repo on GitHub.com. Why should we do this in `hub` through the API then?

**Comment by mislav** [comment]
@hkdobrev From what I can see while inspecting our API validation, it looks like a user can only create a repository *without* a team if they have adminable rights on the org. Any other users will have to supply `team_id`.

Does this make sense?

**Comment by hkdobrev** [comment]
@mislav Fair enough. But I can do that on the web for my organisations as well as any other core dev in the team. Why not be able to do it via `hub`?

**Comment by mislav** [comment]
Have you tried it recently? The API has changed since you reported this in 2013.

**Comment by hkdobrev** [comment]
@mislav You are indeed correct! This is now possible in an organisation where you have admin rights. Great!
